include/nccl_ofi_config_bottom.h

```diff
@@ -11,6 +11,10 @@
 
 #define NCCL_OFI_EXPORT_SYMBOL __attribute__((visibility("default")))
 
+#ifndef __cplusplus
+#define static_assert _Static_assert
```

Comment:
you need an equivalent for C++.  You can't just turn off sanity checks..

Reply:
static_assert is a proper C++ keyword and this doesn't turn off sanity checks under C++, it just adds an alias for all-lowercase `static_assert` to alias to `_Static_assert` under C. C++ Doesn't require that alias, which is why it's C specific...

https://en.cppreference.com/w/cpp/language/static_assert